### PR TITLE
fix(external_api): JS error in removeListeners and simplifies dispose

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

File: modules/API/external/external_api.js [hunk 1]
```diff
@@ -319,16 +319,9 @@ class JitsiMeetExternalAPI extends EventEmitter {
...
-
         this.postis.destroy();
-        if (frame) {
-            frame.src = 'about:blank';
-        }
-        window.setTimeout(() => {
-            this.iframeHolder.removeChild(this.frame);
-            this.iframeHolder.parentNode.removeChild(this.iframeHolder);
-        }, 10);
+        this.removeAllListeners();
```
Comment (saghul) [inline modules/API/external/external_api.js]: Since you are refactoring this, can you add some protection for double-disposal?

Comment (hristoterezov) [inline modules/API/external/external_api.js]: Done!

File: modules/API/external/external_api.js [hunk 2]
```diff
@@ -319,16 +319,13 @@ class JitsiMeetExternalAPI extends EventEmitter {
      * @returns {void}
      */
     dispose() {
-        const frame = document.getElementById(this.frameName);
-
-        this.postis.destroy();
-        if (frame) {
-            frame.src = 'about:blank';
+        if (this.postis) {
+            this.postis.destroy();
```
Comment (saghul) [inline modules/API/external/external_api.js]: maybe reset postis to null so another call to dispose doesn't do anything

Comment (hristoterezov) [inline modules/API/external/external_api.js]: @saghul Done!

**Pull request merged**